docs/file-formats/kmx-file-format.md

```diff
@@ -22,24 +22,24 @@ source files.
-|24 |  32  | cxStoreArray       | number of `COMP_STORE` entries in `COMP_KEYBOARD.dpStoreArray`                    |
-|28 |  32  | cxGroupArray       | number of `COMP_GROUP` entries in `COMP_KEYBOARD.dpGroupArray`                    |
-|32 |  32  | dpStoreArray       | offset of first `COMP_STORE` entry                                                |
-|36 |  32  | dpGroupArray       | offset of first `COMP_GROUP` entry                                                |
-|40 |  32  | StartGroup_ANSI    | deprecated; index of starting non-Unicode `COMP_GROUP`, `0xFFFFFFFF` means unused |
-|44 |  32  | StartGroup_Unicode | index of starting Unicode `COMP_GROUP`, `0xFFFFFFFF` means unused                 |
-|48 |  32  | dwFlags            | global flags for the keyboard, see description below                              |
-|52 |  32  | dwHotKey           | default hotkey for keyboard, from [`&Hotkey`], see description below              |
-|56 |  32  | dpBitmapOffset     | offset of keyboard icon, `0x0` if not present                                     |
-|60 |  32  | dwBitmapSize       | size in bytes of keyboard icon, `0x0` if not present                              |
+| ∆ | Bits | Name                 | Description                                                                       |
+|---|------|----------------------|-----------------------------------------------------------------------------------|
+| 0 |  32  | `dwIdentifier`       | `KXTS` 0x5354584B, `kmx.FILEID_COMPILED`                                          |
+| 4 |  32  | `dwFileVersion`      | minimum Keyman version that can read the file, `KMX_Version` enum                 |
+| 8 |  32  | `dwCheckSum`         | deprecated in 16.0, `0x0`; CRC32 checksum of entire file                          |
+|12 |  32  | `KeyboardID`         | deprecated in 10.0, `0x0`; Windows LANGID for keyboard                            |
+|16 |  32  | `IsRegistered`       | deprecated in 10.0, `0x1`; registration status of compiler                        |
+|20 |  32  | `version`            | deprecated, `0x0`; was version of keyboard, see `TSS_KEYBOARDVERSION`             |
+|24 |  32  | `cxStoreArray`       | number of `COMP_STORE` entries in `COMP_KEYBOARD.dpStoreArray`                    |
+|28 |  32  | `cxGroupArray`       | number of `COMP_GROUP` entries in `COMP_KEYBOARD.dpGroupArray`                    |
+|32 |  32  | `dpStoreArray`       | offset of first `COMP_STORE` entry                                                |
+|36 |  32  | `dpGroupArray`       | offset of first `COMP_GROUP` entry                                                |
+|40 |  32  | `StartGroup_ANSI`    | deprecated; index of starting non-Unicode `COMP_GROUP`, `0xFFFFFFFF` means unused |
+|44 |  32  | `StartGroup_Unicode` | index of starting Unicode `COMP_GROUP`, `0xFFFFFFFF` means unused                 |
```

Comment:
Making sure that I'm parsing the docs correctly - this index lies within the range specified by `dpGroupArray`, right?

Is the index specified here based on the index _within the file_ or the index _offset within (the extracted) `dpGroupArray`_?  That part isn't clear here, and I could see there being arguments for either implementation.

Reply:
> Is the index specified here based on the index _within the file_ or the index _offset within (the extracted) `dpGroupArray`_? That part isn't clear here, and I could see there being arguments for either implementation.

I don't understand the difference you are making here? This is an index, not an offset. So it's the nth item in the array.

Reply:
Oh.  And here I was thinking it was a pointer-style offset (counting bytes/chars/similar), rather than an array index.